cmd/etcd-launcher/main.go

```diff
@@ -130,6 +214,139 @@ func etcdCmd(config *envConfig) []string {
 			"--experimental-corrupt-check-time=10m",
 		}...)
 	}
-
 	return cmd
 }
+
+func (e *etcdCluster) getClient() error {
+	if e.client != nil {
```

Comment:
I don't understand this one, so when the client is not nil you are returning nil, shouldn't you return the client here?

Reply:
`client` is set in the `etcdCluster` struct. If it's nil, I create it and assign it to e.client instead of passing it around.